checks.d/spark.py

```diff
@@ -211,23 +208,103 @@ def check(self, instance):
         self._spark_rdd_metrics(spark_apps, tags)
 
         # Report success after gathering all metrics from the ApplicationMaster
-        if running_apps:
-            app_id, (app_name, tracking_url) = running_apps.items()[0]
+        if spark_apps:
+            app_id, (app_name, tracking_url) = spark_apps.items()[0]
             am_address = self._get_url_base(tracking_url)
 
             self.service_check(SPARK_SERVICE_CHECK,
                 AgentCheck.OK,
                 tags=['url:%s' % am_address],
                 message='Connection to ApplicationMaster "%s" was successful' % am_address)
 
-    def _get_running_spark_apps(self, rm_address):
+    def _get_running_apps(self, instance, tags):
+        '''
+        Figures out what mode we're in and fetches running apps
+        '''
+        cluster_mode = instance.get(SPARK_CLUSTER_MODE)
```

Comment:
We should probably default to `SPARK_YARN_MODE` - after maybe printing a warning. This is to avoid any backward compatibility breakage for existing customers.